doc/api/stream.md

```diff
@@ -61,6 +61,15 @@ Stream instances are switched into object mode using the `objectMode` option
 when the stream is created. Attempting to switch an existing stream into
 object mode is not safe.
 
+The object mode indicates that the stream accepts and/or emits values which are
+not limited to `Buffer` or strings. Be aware that streams which are in object
+mode can be connected to streams not in the object mode but an exception will
```

Comment:
If “connected” here means using `.pipe()`, then this should be made explicit, I think – otherwise it’s not clear what the word means in this context.